### PR TITLE
EL-1776 add security and robots endpoints

### PR DESCRIPTION
**Pull request by willc-work**
## What does this pull request do?

Adds 2 endpoints for NCSC to check but it is also good practice to have these
add security.txt from moj standard template
https://github.com/ministryofjustice/security-guidance/blob/3245a63a5e3816a17d253eadcf81b4f2ee97663f/docs/security.txt#L4
add robots to allow crawlers in prod only

## Any other changes that would benefit highlighting?

Intentionally left blank.

## Checklist

- [ ] Provided JIRA ticket number in the title, e.g. "LGA-152: Sample title"

### CODE REVIEW

File: fala/apps/adviser/views.py [hunk 1]
```diff
@@ -4,11 +4,37 @@
...
 from django.urls import resolve, reverse
 from django.views.generic import TemplateView, ListView
+from django.http import HttpResponse
+import os
 
 from .forms import AdviserSearchForm, AdviserRootForm
 from .laa_laa_paginator import LaaLaaPaginator
 from .regions import Region
 
+BASE_DIR = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
```
Comment (willc-work) [inline fala/apps/adviser/views.py:14]: should this be in settings.py?

File: fala/urls.py [hunk 2]
```diff
@@ -2,7 +2,15 @@
...
 
 from django.urls import re_path
-from adviser.views import AccessibilityView, AdviserView, SearchView, PrivacyView, CookiesView
+from adviser.views import (
+    AccessibilityView,
+    AdviserView,
+    SearchView,
+    PrivacyView,
+    CookiesView,
+    robots_txt,
```
Comment (willc-work) [inline fala/urls.py:11]: i dont know if these need to match the style of the other views here but they feel like different pages so it doesnt seem too wrong but happy to amend if needed

File: fala/apps/adviser/views.py (hunk 1 above)
Comment (psweeting1) [reply fala/apps/adviser/views.py:14]: Yes, this should be in settings / base.py 

So you can do something like this `settings.BASE_DIR` or whatever to access the BASE_DIR variable

File: fala/apps/adviser/views.py [hunk 3]
```diff
@@ -4,11 +4,36 @@
...
+    if environment == "production":
+        content = "User-agent: *\nDisallow:\n"  # Allow all
+    else:
+        content = "User-agent: *\nDisallow: /\n"  # Disallow all for staging/UAT
+
+    return HttpResponse(content, content_type="text/plain")
+
+
+def security_txt(request):
+    file_path = os.path.join(BASE_DIR, "public", "security.txt")
```
Comment (psweeting1) [inline fala/apps/adviser/views.py:29]: import and link BASE_DIR to the settings variable.

File: fala/apps/adviser/views.py [hunk 4]
```diff
@@ -4,11 +4,36 @@
...
+import os
 
 from .forms import AdviserSearchForm, AdviserRootForm
 from .laa_laa_paginator import LaaLaaPaginator
 from .regions import Region
 
+BASE_DIR = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+
+
+def robots_txt(request):
```
Comment (psweeting1) [inline fala/apps/adviser/views.py:17]: I was thinking about consistency as the imports in the URL side doesn't match or other URL's

Up to you, but we could put it into a class to keep the consistency. Happy to discuss your thought on this.
```
class RobotsTxtView(View):
    def get(self, request):
        environment = os.getenv("ENVIRONMENT", "development").lower()

        if environment == "production":
            content = "User-agent: *\nDisallow:\n"  # Allow all
        else:
            content = "User-agent: *\nDisallow: /\n"  # Disallow all for staging/UAT

        return HttpResponse(content, content_type="text/plain")
```

File: fala/urls.py (hunk 2 above)
Comment (psweeting1) [reply fala/urls.py:11]: This is fine, it looks odd as you're not importing a class but a function instead.

**Review by psweeting1 — APPROVED** [review]
LGTM

**Pull request merged**